pkg/controllers/provisioning/launcher.go

```diff
@@ -124,7 +124,7 @@ func (l *Launcher) verifyResourceLimits(ctx context.Context, provisioner *v1alph
 		return err
 	}
 	for resourceName, usage := range provisionerLatest.Status.Resources {
-		if limit, ok := provisionerLatest.Spec.Limits.Resources[resourceName]; ok {
+		if limit, ok := provisioner.Spec.Limits.Resources[resourceName]; ok {
```

Comment:
Sorry to keep layering on comments. One more optional idiomatic improvement:

```
latest := &v1alpha5.Provisioner{}
if err := l.KubeClient.Get(ctx, client.ObjectKeyFromObject(provisioner), latest); err != nil {
	return err
}
if err := provisioner.Spec.Limits.ExceededBy(latest.Status.Resources); err != nil {
  return err
}
return nil
```

Reply:
Nice, thanks! Done.